app/components/NavBar.js

```diff
@@ -50,6 +50,27 @@ const NavBar = () => {
   const { isOpen, onOpen, onClose } = useDisclosure();
   const navRef = useRef(null);
   const pathname = usePathname();
+  const [searchProjectsText, setSearchProjectsText] = useState('')
+
+  const router = useRouter();
+
+  const handleSearch = (e) => {
+    e.preventDefault();
+
+    // Redirect to the /project URL with the search term as a query parameter
+    router.push(`/projects?search=${encodeURIComponent(searchProjectsText)}`);
```

Comment:
@D-Lite I think you should use `router.replace()` here. So we don't end up populating the history stack on every search change 

You can be explicit about it too by including the `shallow` property and setting it to `true` like so:

```js
router.replace(`/projects?search=${encodeURIComponent(searchProjectsText)}`, undefined, {shallow: true})
```

More on that [here](https://nextjs.org/docs/pages/api-reference/functions/use-router#routerreplace)

Reply:
Yes, good performance improvement. 

Thanks @kaf-lamed-beyt

Reply:
Anytime, chief! 🍷

Reply:
Thanks fellas @D-Lite @kaf-lamed-beyt